Fixes #9121: better readme for the github project page

dmitri-d opened the pull request.

dmitri-d committed:
Fixes #9121: better readme for the github project page

ohadlevy commented on README.md:
```diff
@@ -0,0 +1,44 @@
+ * Puppet - Any Puppet server from 0.24.x
+ * Puppet CA - Manage certificate signing, cleaning and autosign on a Puppet CA server
+ * Realm - Manage host registration to a realm (e.g. FreeIPA)
+ * TFTP - any UNIX based tftp server
```
libvirt?

reference to other plugins? / talk about plugin arch etc?

dmitri-d replied:
Is it worth talking about libvirt here? Install docs (referenced below) cover that install scenario...

ohadlevy commented on README.md:
```diff
@@ -0,0 +1,44 @@
+
+# For Developers
+* [API Reference](http://projects.theforeman.org/projects/smart-proxy/wiki/API)
+* Smart Proxy Plugin development [how-to] (http://projects.theforeman.org/projects/foreman/wiki/How_to_Create_a_Smart-Proxy_Plugin)
```
oh, missed it.

logicminds commented:
I actually started a similar development doc but haven’t contributed this yet.  Here is what I wrote when I was working on a some BMC stuff.  I can submit a PR later today.  Or feel free to just add what I wrote to this PR.

https://gist.github.com/logicminds/d387d27d273374797c3b

dLobatog commented on README.md:
```diff
@@ -0,0 +1,44 @@
+[![Issue Stats](http://issuestats.com/github/theforeman/smart-proxy/badge/pr)](http://issuestats.com/github/theforeman/smart-proxy)
+[![Support IRC channel](https://kiwiirc.com/buttons/irc.freenode.net/theforeman.png)](https://kiwiirc.com/client/irc.freenode.net/?#theforeman)
+
+[Smart Proxy](http://projects.theforeman.org/projects/smart-proxy/wiki) is a free open source project that provides restful API to subsystems such as DNS, DHCP, etc, for a higher level orchestration tools such as [Foreman](https://github.com/theforeman/foreman). 
```
`for higher level orchestration tools`

dLobatog commented on README.md:
```diff
@@ -0,0 +1,44 @@
+
+# Special thanks
+
+The original author of this project is [Ohad Levy](http://github.com/ohadlevy)
```
If you meant to put this in separate lines I think you have to put an empty line in between these.

domcleal commented on README.md:
```diff
@@ -0,0 +1,44 @@
+Read the [Smart Proxy Installation section](http://theforeman.org/manuals/1.8/index.html#4.3.1SmartProxyInstallation) of the manual.
+
+# Configuration
+Read the [Smart Proxy Settings section](http://theforeman.org/manuals/1.8/index.html#4.3.2SmartProxySettings) of the manual.
```
use `latest` rather than versions I'd say

dmitri-d commented:
@logicminds: updated  development docs at http://projects.theforeman.org/projects/foreman/wiki/How_to_Create_a_Smart-Proxy_Plugin. @domcleal, @eLobato: fixed issues you pointed out.

dLobatog commented:
Merged as c08eb9a80d33655279ad4cc6039f0299ccb14e3b, thanks @witlessbird !
